crates/driver/src/infra/simulator/enso/mod.rs

```diff
@@ -36,7 +67,8 @@ impl Enso {
                 data: tx.input.into(),
                 value: tx.value.into(),
                 gas_limit: GAS_LIMIT,
-                block_number: None,
+                block_number,
```

Comment:
Side note: Specifically setting the block number to the latest one we have state for might fix https://github.com/cowprotocol/infrastructure/issues/946 as well.